js/main.js

```diff
@@ -527,3 +527,139 @@ var xint = setInterval(function() {
   addEvent(document.body, 'mousedown', makeEditable);
   clearInterval(xint);
 }, 200);
+
+function generateDonationUrl(address, amountBtc, message) {
+    var result = [
+        address
+    ];
+
+    amountBtc = parseFloat(amountBtc);
+
+    if (!isNaN(amountBtc)) {
```

Comment:
> Check that amountBtc is a number first.
> For example, isNaN('') returns false because it first tries to convert to number, than checks NaN.

A check for being a number is already done using parseFloat:
```
amountBtc = parseFloat('') // NaN
isNaN(amountBtc) // true
```

`> 0` check is not really necessary as this transaction doesn't make sense.  As I will make a few changes, I'll add it.